### PR TITLE
Added tests for CreateUserAccount, BulkEnrollment and UpdateUserAccount API views

### PR DESCRIPTION
**Pull request by OmarIthawi**
**Note:** The Bulk Enrollment API tests are borrowed from and modified from upstream:
  - https://github.com/edx/edx-platform/blob/1990dc7/lms/djangoapps/bulk_enroll/tests/test_views.py
  - edx#15006

### CODE REVIEW

**Comment by OmarIthawi** [comment]
[(16) Appsembler API: Implement Test](https://trello.com/c/QMjvZFj6/1315-16-appsembler-api-implement-test)

File: lms/djangoapps/appsembler_api/tests/test_bulk_enrollments.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+            }
+        }
+
+        manual_enrollments = ManualEnrollmentAudit.objects.all()
+        self.assertEqual(manual_enrollments.count(), 1)
+        self.assertEqual(manual_enrollments[0].state_transition, UNENROLLED_TO_ENROLLED)
+        res_json = json.loads(response.content)
+        self.assertEqual(res_json, expected)
+
+        # Check the outbox
```
[the remark this answers is not in the document]
Comment (bryanlandia) [reply lms/djangoapps/appsembler_api/tests/test_bulk_enrollments.py]: ... and then an additional method to exercise the condition where `'email_students'` is true ... and `len(mail.outbox) == 1`

File: lms/djangoapps/appsembler_api/tests/test_create_user_account.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+@ddt.ddt
+@patch('appsembler_api.views.CreateUserAccountView.authentication_classes', [])
+@patch('appsembler_api.views.CreateUserAccountView.permission_classes', [AllowAny])
+@override_settings(APPSEMBLER_FEATURES={
+    'SKIP_LOGIN_AFTER_REGISTRATION': False,
+})
+class CreateUserAccountViewTests(TestCase):
+    def setUp(self):
+        self.url = reverse('create_user_account_api')
+        self.sample_user_date = {
```
Comment (bryanlandia) [inline lms/djangoapps/appsembler_api/tests/test_create_user_account.py]: nitpick:  Why `_date`?  Supposed to be `_data`?

File: lms/djangoapps/appsembler_api/tests/test_permissions.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+from appsembler_api import views as api_views
+
+
+def get_api_classes():
+    api_classes = []
+
+    for member_name in dir(api_views):
+        member = getattr(api_views, member_name)
+        if isclass(member):
+            if member.__module__ == api_views.__name__:
```
Comment (bryanlandia) [inline lms/djangoapps/appsembler_api/tests/test_permissions.py]: Just curious... why this member.__module__ check?  Doesn't your for...in loop already take care of this restriction to `api_views`?

Comment (bryanlandia) [reply lms/djangoapps/appsembler_api/tests/test_permissions.py]: Oh,.... or does this prevent including imported classes?

File: lms/djangoapps/appsembler_api/tests/test_update_user_account.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    name = attr.ib()
+    valid = attr.ib(default=None)
+    invalid = attr.ib(default=None)
+
+
+@ddt.ddt
+@patch('appsembler_api.views.UpdateUserAccount.authentication_classes', [])
+@patch('appsembler_api.views.UpdateUserAccount.permission_classes', [AllowAny])
+class CreateUserAccountViewTests(TestCase):
+    USER_FIELDS = [
```
Comment (bryanlandia) [inline lms/djangoapps/appsembler_api/tests/test_update_user_account.py]: Doesn't look like `USER_FIELDS` is used

File: lms/djangoapps/appsembler_api/views.py [hunk 5]
```diff
@@ -284,30 +284,31 @@ def post(self, request):
...
+            user = User.objects.get(
+                Q(username=data['user_lookup']) | Q(email=data['user_lookup'])
+            )
+        except User.DoesNotExist:
+            return Response({
+                "user_not_found": "The user for the Given username or email doesn't exists",
+            }, status=404)
+        except User.MultipleObjectsReturned:
+            return Response({
+                "lookup_error": "Two users have been found wiht the provided user_lookup",
```
Comment (bryanlandia) [inline lms/djangoapps/appsembler_api/views.py]: nitpick:  Should be "found with"

File: lms/djangoapps/appsembler_api/tests/test_permissions.py (hunk 3 above)
Comment (OmarIthawi) [reply lms/djangoapps/appsembler_api/tests/test_permissions.py]: > Oh,.... or does this prevent including imported classes?

You got it right!

File: lms/djangoapps/appsembler_api/tests/test_update_user_account.py (hunk 4 above)
Comment (OmarIthawi) [reply lms/djangoapps/appsembler_api/tests/test_update_user_account.py]: Terminated.

**Comment by OmarIthawi** [comment]
Thanks @bryanlandia for the review. I've addressed all of your comments, except for the `test_bulk_enrollments` module, since it's mostly copied from master and I don't feel like it's worth spending much effort on it given that this is already planned to be [retired in Hawthorne](https://trello.com/c/Ko5bm1EB/2726-move-the-customers-to-the-updated-hawthorne-bulk-enrollment-api).

Merging now.

**Pull request merged**